plugins/woocommerce/src/Internal/Admin/Notes/WooSubscriptionsNotes.php

```diff
@@ -301,9 +301,10 @@ public function add_or_update_subscription_expiring( $subscription ) {
 
 				foreach ( (array) $bump_thresholds as $bump_threshold ) {
 					if ( ( $note_days_until_expiration > $bump_threshold ) && ( $days_until_expiration <= $bump_threshold ) ) {
-						$note->delete();
-						$note = false;
-						continue;
+						if ( $note ) {
+							$note->delete();
+							$note = false;
+						}
```

Comment:
Needing to exit a `foreach` loop does not seem to be a good pattern: either we need to change the loop type or let it loop through all elements (`array( 60, 45, 20, 7, 1 );`). I think it is ok to skip` continue` (or `break`) if we use `foreach`.

Reply:
I think it would be clearer to just use `break` to exit the `foreach`. There is no reason to keep on looping once there is no possibility that we will need to do anything else.

Reply:
We could refactor the logic out into a function such as `should_note_be_bumped_to_the_top_of_inbox( $note, $days_until_expiration )`. 

Inside that function, we would check the expiration listed in the note against the thresholds to determine if it is in need of bumping.

This would make the code a lot more readable.

We could also write some tests for that function.